apps/storefront/src/pages/QuickOrder/index.quickadd.test.tsx

```diff
@@ -0,0 +1,768 @@
+
+const preloadedState = { company: approvedB2BCompany, storeInfo: storeInfoWithDateFormat };
+
+beforeEach(() => {
+  /* @ts-expect-error This object is not complete, it only includes the properties required for this test file */
+  window.b2b = { callbacks: { dispatchEvent: vi.fn() } };
+});
+
+afterEach(() => {
+  // @ts-expect-error Removing the b2b object to avoid conflicts in other tests
+  delete window.b2b;
+
+  Cookies.remove('cartId');
+});
+
+it('renders the quick add section', () => {
+  renderWithProviders(<QuickOrderPad />, { preloadedState });
+
+  expect(screen.getByText('Quick add')).toBeInTheDocument();
+});
+
+it('increases the number of input rows when clicking -show more rows- button', async () => {
+  renderWithProviders(<QuickOrderPad />, { preloadedState });
+
+  const showMoreRowsButton = screen.getByRole('button', { name: 'Show more rows' });
+
+  expect(screen.getAllByRole('textbox', { name: 'SKU#' })).toHaveLength(3);
+  expect(screen.getAllByRole('spinbutton', { name: 'Qty' })).toHaveLength(3);
+
+  await userEvent.click(showMoreRowsButton);
+
+  expect(screen.getAllByRole('textbox', { name: 'SKU#' })).toHaveLength(6);
+  expect(screen.getAllByRole('spinbutton', { name: 'Qty' })).toHaveLength(6);
+});
+
+it('adds the skus and the quantities to the cart when clicking on the -add to cart- button', async () => {
+  const getVariantInfoBySkus = vi.fn();
+
+  const variantInfo = buildVariantInfoWith({
+    variantSku: 'S-123',
+    minQuantity: 0,
+    purchasingDisabled: '0',
+    isStock: '1',
+  });
+
+  when(getVariantInfoBySkus)
+    .calledWith(expect.stringContaining('variantSkus: ["S-123","S-456"]'))
+    .thenDo(() => buildVariantInfoResponseWith({ data: { variantSku: [variantInfo] } }));
+
+  const createCartSimple = vi.fn();
+
+  when(createCartSimple)
+    .calledWith({
+      createCartInput: {
+        lineItems: [
+          {
+            quantity: 2,
+            productEntityId: Number(variantInfo.productId),
+            variantEntityId: Number(variantInfo.variantId),
+            selectedOptions: {
+              multipleChoices: [],
+              textFields: [],
+            },
+          },
+        ],
+      },
+    })
+    .thenReturn({
+      data: {
+        cart: { createCart: { cart: { entityId: 'de435179-9b4b-4fa4-b609-34d948d04783' } } },
+      },
+    });
+
+  server.use(
+    graphql.query('GetVariantInfoBySkus', ({ query }) =>
+      HttpResponse.json(getVariantInfoBySkus(query)),
+    ),
+    graphql.query('getCart', () => HttpResponse.json<GetCart>({ data: { site: { cart: null } } })),
+    graphql.mutation('createCartSimple', ({ variables }) =>
+      HttpResponse.json(createCartSimple(variables)),
+    ),
+  );
+
+  renderWithProviders(<QuickOrderPad />, { preloadedState });
+
+  const [skuInput, missingSkuInput] = screen.getAllByRole('textbox', { name: 'SKU#' });
+  const [qtyInput, missingQtyInput] = screen.getAllByRole('spinbutton', { name: 'Qty' });
```

Comment:
fun use of de-structoring 

`missingSkuInput` is a bit confusing as we're naming the input after a 'meta' quality of a value that we plan to enter into it

rather than the norm of naming an input based on its label/purpose

it's probably worth giving the special behaviour for "missing" products its own test